### PR TITLE
Decoder

### PR DESCRIPTION
**Pull request by uslsteen**
1. Implement initial decoder codegen.

Relates #5

### CODE REVIEW

**Review by Dimanteo — APPROVED** [review]
LGBT

File: include/executor/executor.hpp [hunk 1]
```diff
@@ -1,6 +1,25 @@
...
 
-namespace sim {}
+#include "isa/instr.hpp"
+#include <unordered_map>
+
+namespace sim {
+
+class Executor final {
+
+  std::unordered_map<int, int> m_exec_instr{};
```
Comment (Dimanteo) [inline include/executor/executor.hpp]: What does this map contain?

**Review by DragonZoom — COMMENTED** [review]
LGTM

File: include/executor/executor.hpp (hunk 1 above)
Comment (uslsteen) [reply include/executor/executor.hpp]: Everything.

**Pull request merged**